### ISSUE TITLE
@Transient properties of base class are never initialized

### ISSUE THREAD

**Issue by SnailAfterRain** [Kotlin/kotlinx.serialization#408]
Don't know about JVM, but in JS if you have a class whose base class has any `@Transient` properties, they're never initialized. Consider this code:

```kotlin
@Serializable
open class Foo {
    @Transient var fooMeta = "foo meta"

    init {
        println("foo init")
    }
}

@Serializable
data class Bar(val name: String, val id: Int) : Foo() {
    @Transient var barMeta = "bar meta"

    init {
        println("bar init")
    }
}


fun test() {
    val json = Json(unquoted = true, indented = true)
    val serializer = Bar.serializer()

    val text = json.stringify(serializer, Bar("Hey", 123))
    val bar = json.parse(serializer, text)

    println(bar)
    println("barMeta ${bar.barMeta}")
    println("fooMeta ${bar.fooMeta}")
}
```
Calling `test` outputs:
```
foo init
bar init
bar init
Bar(name=Hey, id=123)
barMeta bar meta
fooMeta undefined
```
As you can see, base constructor isn't called either. Removing `@Transient` works just fine, but it kinda defeats the whole purpose of what i'm trying to do. 

Any way around it?

**Comment by sandwwraith** [comment]
This looks like a compiler bug which is fixed already in `master`. Next release should not have this problem.

**Comment by SnailAfterRain** [comment]
If anyone really needs this behavior right now and can't wait for next release, you can use something like this 

```kotlin
@Serializable
open class Foo {
    @Transient val initGuard = true
    @Transient var fooMeta = "foo meta"

    init {
        println("foo init")
    }

    inline fun <reified S : Foo> initIfNeeded() {
        if(initGuard == undefined)
            S::class.js.asDynamic().call(this)
    }
}

@Serializable
class Bar(val name: String) : Foo() {
    @Transient var barMeta = "bar meta"

    init {
        initIfNeeded<Foo>()
        println("bar init")
    }
}
```
You'll need to call `initIfNeeded` in every init block of your derived classes with it's super type

**Comment by SnailAfterRain** [comment]
Fixed in 0.11.0
